src/main/resources/assets/js/application/common/filters/tdprJobProfileFilter.js

```diff
@@ -0,0 +1,36 @@
+define(['application/common/tdprCommonModule'], function (tdprCommonModule) {
+    tdprCommonModule.filter('jobProfileFilter', function () {
+        return function (toFilter, jobProfile) {
+            if (jobProfile === null || jobProfile === undefined) {
```

Comment:
The `object === null || object === undefined` check is used repeatedly in the code - perhaps it's worth to extract a function?